### PR TITLE
Fix NTAuthentication.MakeSignature/VerifySignature on Linux

### PR DESCRIPTION
**Pull request by filipnavara**
Contributes to #65678

### CODE REVIEW

**Comment by filipnavara** [comment]
The Windows tests fail because `VerifySignature` in NegotiateStreamPal.Windows doesn't return the content which is fixable. Not sure why the macOS tests fail since they previously passed on my machine, have to dig in a bit deeper.

File: src/libraries/Common/src/System/Net/Security/NegotiateStreamPal.Unix.cs [hunk 1]
```diff
@@ -618,21 +619,8 @@ internal static int VerifySignature(SafeDeleteContext securityContext, byte[] bu
...
-            {
-                output[0] = (byte)((resultSize) & 0xFF);
-                output[1] = (byte)(((resultSize) >> 8) & 0xFF);
-                output[2] = (byte)(((resultSize) >> 16) & 0xFF);
-                output[3] = (byte)(((resultSize) >> 24) & 0xFF);
-            }
-
-            return resultSize + 4;
+            output = GssWrap(gssContext.GssContext, false, new ReadOnlySpan<byte>(buffer, offset, count));
+            return output.Length;
```
Comment (wfurt) [inline src/libraries/Common/src/System/Net/Security/NegotiateStreamPal.Unix.cs:623]: The change seems ok to me. The surprise is that I don't see any changes callers of this function. 
That suggest that either 1) it is not used or 2) we broke something that is not covered by tests.

Comment (filipnavara) [reply src/libraries/Common/src/System/Net/Security/NegotiateStreamPal.Unix.cs:623]: It is not covered by tests, it is only used by SMTP authentication.

File: src/libraries/System.Net.Security/tests/UnitTests/NTAuthenticationTests.cs [hunk 2]
```diff
@@ -16,6 +18,7 @@ public class NTAuthenticationTests
 
         private static NetworkCredential s_testCredentialRight = new NetworkCredential("rightusername", "rightpassword");
         private static NetworkCredential s_testCredentialWrong = new NetworkCredential("rightusername", "wrongpassword");
+        private static byte[] s_Hello = new byte[] { (byte)'H', (byte)'e', (byte)'l', (byte)'l', (byte)'o' };
```
Comment (filipnavara) [inline src/libraries/System.Net.Security/tests/UnitTests/NTAuthenticationTests.cs:21]: Are the new UTF-8 literals in C# yet? Does runtime use new enough compiler?

Comment (am11) [reply src/libraries/System.Net.Security/tests/UnitTests/NTAuthenticationTests.cs:21]: [Yup](https://github.com/dotnet/runtime/pull/68334)

Comment (filipnavara) [reply src/libraries/System.Net.Security/tests/UnitTests/NTAuthenticationTests.cs:21]: Thanks 👍 I will update the code to use it.

**Review by wfurt — APPROVED** [review]
LGTM

**Comment by wfurt** [comment]
System.IO.Pipes.Tests failure is unrelated.

**Pull request merged**